crypto/x509/x509_vfy.c

```diff
@@ -124,20 +122,16 @@ static int lookup_cert_match(X509 **result, X509_STORE_CTX *ctx, X509 *x)
     ERR_pop_to_mark();
     if (certs == NULL)
         return -1;
+
     /* Look for exact match */
     for (i = 0; i < sk_X509_num(certs); i++) {
         xtmp = sk_X509_value(certs, i);
         if (X509_cmp(xtmp, x) == 0)
             break;
         xtmp = NULL;
     }
-    ret = xtmp != NULL;
-    if (ret) {
-        if (!X509_up_ref(xtmp))
-            ret = -1;
-        else
-            *result = xtmp;
-    }
+    *result = xtmp;
+    ret = *result == NULL ? 0 : X509_up_ref(*result) ? 1 : -1;
```

Comment:
Do the nested ternarys really make the code more readable and maintainable?

Reply:
Yeah, I do not see them readable either - At least I would use parentheses around the internal ternary but IMO the original is more readable.

Reply:
In general, I like conditional expressions (using the `? :` ternary operator) for their conciseness.
Yet I agree that at least here the code is less readable because the conditions are nested.
I'm going to revert this type of changes.